internal/component/kafka/consumer.go

```diff
@@ -35,28 +36,110 @@ type consumer struct {
 
 func (consumer *consumer) ConsumeClaim(session sarama.ConsumerGroupSession, claim sarama.ConsumerGroupClaim) error {
 	b := consumer.k.backOffConfig.NewBackOffWithContext(session.Context())
+	var messages []*sarama.ConsumerMessage
+	messages = make([]*sarama.ConsumerMessage, 0)
+	isBulkSubscribe := consumer.k.checkBulkSubscribe(claim.Topic())
 	for message := range claim.Messages() {
+		if isBulkSubscribe {
+			if len(messages) < consumer.k.bulkSubscribeConfig.MaxBulkCount {
```

Comment:
MaxBulkCount here works as the minimum count after which you break out of the loop ...